core/server/api/shared/pipeline.js

```diff
@@ -156,6 +156,9 @@ const pipeline = (apiController, apiUtils) => {
                 })
                 .then(() => {
                     return frame.response;
+                })
+                .catch((err) => {
+                    throw {err, docName, method};
```

Comment:
This rethrow is needed here to be able to pass docName & method information higher up to `http` handler. It's ugly but couldn't think of any better way to attach that information to `req`. Maybe you have some better ideas on mind @kirrg001 ?